docs/fundamentals/orchestrate-resources.md

```diff
@@ -0,0 +1,212 @@
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var postgres = builder.AddPostgres("postgres");
+var postgresdb = postgres.AddDatabase("postgresdb");
+
+var migration = builder.AddProject<Projects.AspireApp_Migration>("migration")
+                       .WithReference(postgresdb)
+                       .WaitFor(postgresdb);
+
+builder.AddProject<Projects.AspireApp_ApiService>("apiservice")
+       .WithReference(postgresdb)
+       .WaitForCompletion(migration);
+```
+
+In the preceding code, the "apiservice" project resource waits for the "migration" project resource to run to completion before starting. The "migration" project resource waits for the "postgresdb" database resource to enter the <xref:Aspire.Hosting.ApplicationModel.KnownResourceStates.Running?displayProperty=nameWithType> state. This can be useful in scenarios where you want to run a database migration before starting the API service, for example.
+
+### Forcing resource start in the dashboard
+
+Waiting for a resource can be bypassed using the "Start" command in the dashboard. Clicking "Start" on a waiting resource in the dashboard instructs it to start immediately without waiting for the resource to be healthy or completed. This can be useful when you want to test a resource immediately and don't want to wait for the app to be in the right state.
+
+## APIs for adding and expressing resources
+
+.NET Aspire [hosting integrations](integrations-overview.md#hosting-integrations) and [client integrations](integrations-overview.md#client-integrations) are both delivered as NuGet packages, but they serve different purposes. While _client integrations_ provide client library configuration for consuming apps outside the scope of the app host, _hosting integrations_ provide APIs for expressing resources and dependencies within the app host. For more information, see [.NET Aspire integrations overview: Integration responsibilities](integrations-overview.md#integration-responsibilities).
+
+## Express container resources
+
+To express a <xref:Aspire.Hosting.ApplicationModel.ContainerResource> you add it to an <xref:Aspire.Hosting.IDistributedApplicationBuilder> instance by calling the <xref:Aspire.Hosting.ContainerResourceBuilderExtensions.AddContainer%2A> method:
+
+### [Docker](#tab/docker)
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var ollama = builder.AddContainer("ollama", "ollama/ollama")
+    .WithBindMount("ollama", "/root/.ollama")
+    .WithBindMount("./ollamaconfig", "/usr/config")
+    .WithHttpEndpoint(port: 11434, targetPort: 11434, name: "ollama")
+    .WithEntrypoint("/usr/config/entrypoint.sh")
+    .WithContainerRuntimeArgs("--gpus=all");
```

Comment:
Don't these need to be converted to snippets?

Reply:
That is ideal, yes! We're still working on getting ALL example source as snippets. Probably need to create an issue.